src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.cs

```diff
@@ -518,6 +518,10 @@ private TypeBuilder(string szName, int genParamPos, TypeBuilder declType)
                         // cannot contain null in the interface list
                         throw new ArgumentNullException(nameof(interfaces));
                     }
+                    if (interfaces[i].IsByRef)
```

Comment:
Reflection.Emit should duplicate type validation like this. It should be done in the type loader in the VM only.

Reply:
This is done in the ref emit pass since it was originally in the `m_module.GetTypeTokenInternal(interfaces[i]);` on line 529. I don't think I should be reworking the Reflection Emit API that much for this particular change.

Reply:
Based on the discussion in https://github.com/dotnet/runtime/issues/43282#issuecomment-726895475 , we wanted to delete the `IsByRef` check in `GetTypeTokenInternal` as part of the fix for that issues.